src/Microsoft.Health.Dicom.Functions/Indexing/ReindexDurableFunction.cs

```diff
@@ -18,33 +21,41 @@ namespace Microsoft.Health.Dicom.Functions.Indexing
     /// </summary>
     public partial class ReindexDurableFunction
     {
-        private readonly ReindexConfiguration _reindexConfig;
+        private readonly ReindexOperationConfiguration _reindexConfig;
         private readonly IReindexStore _reindexStore;
         private readonly IInstanceReindexer _instanceReindexer;
         private readonly IAddExtendedQueryTagService _addExtendedQueryTagService;
         private readonly IInstanceStore _instanceStore;
         private readonly IExtendedQueryTagStore _extendedQueryTagStore;
+        private readonly ISchemaManagerDataStore _schemaManagerDataStore;
+        private readonly SchemaInformation _schemaInformation;
 
         public ReindexDurableFunction(
-            IOptions<IndexingConfiguration> configOptions,
+            IOptions<DicomFunctionsConfiguration> configOptions,
             IAddExtendedQueryTagService addExtendedQueryTagService,
             IReindexStore reindexStore,
             IInstanceStore instanceStore,
             IInstanceReindexer instanceReindexer,
-            IExtendedQueryTagStore extendedQueryTagStore)
+            IExtendedQueryTagStore extendedQueryTagStore,
+            ISchemaManagerDataStore schemaManagerDataStore,
+            SchemaInformation schemaInformation)
         {
             EnsureArg.IsNotNull(configOptions, nameof(configOptions));
             EnsureArg.IsNotNull(reindexStore, nameof(reindexStore));
             EnsureArg.IsNotNull(instanceReindexer, nameof(instanceReindexer));
             EnsureArg.IsNotNull(addExtendedQueryTagService, nameof(addExtendedQueryTagService));
             EnsureArg.IsNotNull(instanceStore, nameof(instanceStore));
             EnsureArg.IsNotNull(extendedQueryTagStore, nameof(extendedQueryTagStore));
-            _reindexConfig = configOptions.Value.Add;
+            EnsureArg.IsNotNull(schemaManagerDataStore, nameof(schemaManagerDataStore));
+            EnsureArg.IsNotNull(schemaInformation, nameof(schemaInformation));
+            _reindexConfig = configOptions.Value.Services.Reindex;
```

Comment:
We have generally injected something like `IOptions<ReindexOperationConfiguration>` elsewhere.